ristretto.sage

```diff
@@ -835,6 +837,40 @@ def testElligator(cls,n):
         else:
             pass # TODO
 
+def testElligatorDeterministic(cls):
+    """These test cases correspond to those in the Decaf377 crate in test_elligator"""
+
+    # Test case inputs were generated beginning with the value
+    # 2873166235834220037104482467644394559952202754715866736878534498814378075613
+    # and then are the x-coordinate of the previous input.
```

Comment:
Rather than using the x-coordinate of the previous input, I think it would be better to use the `s` value that the output point encodes to.  The reason is that then it's possible to write the test using only externally-visible data; if the x-coordinate is used, building the test inputs requires breaking through the construction's abstraction boundary.